### PR TITLE
Feature/mktar

### PR DESCRIPTION
**Pull request by andre-merzky**
Support tarball creation to simplify higher level codes (RP, RX)

### CODE REVIEW

**Comment by andre-merzky** [comment]
@mturilli : do you have time for a quick review?  This optimizes repex (among others...)

File: src/radical/utils/misc.py [hunk 1]
```diff
@@ -848,5 +850,46 @@ def rec_makedir(target):
             raise
 
 
+# ------------------------------------------------------------------------------
+#
+def mktar(tarname, fnames=None, data=None):
+    '''
+    Create a tarfile at the given `tarname`, and pack all files given in
+    `fnames` into it, and lso pack any `data` blobs.
```
Comment (mturilli) [inline src/radical/utils/misc.py]: lso->also

File: src/radical/utils/misc.py [hunk 2]
```diff
@@ -848,5 +850,46 @@ def rec_makedir(target):
...
+
+    `fnames` is expected to be list, where each element is either a string
+    pointing to a file to be added under that name, or a tuple where the first
+    element points again to the file to be packed, and the second element
+    specifies the name under which the file should be packed into the archive.
+
+    `data` is expected to
```
Comment (mturilli) [inline src/radical/utils/misc.py]: spurious \n

File: src/radical/utils/misc.py [hunk 3]
```diff
@@ -848,5 +850,46 @@ def rec_makedir(target):
...
+    be a list of tuples, where the first element is a set of bytes comprising
+    the data to be written into the archive, and the second element again
+    specifies the name of the tarred file.
+
+    Note that this method always create gzip'ed tarfiles, but will never change
+    the `tarname` to reflect that.
```
Comment (mturilli) [inline src/radical/utils/misc.py]: Should bzip2 be the default compression seeing the (sometimes vast) superior efficiency?

File: src/radical/utils/misc.py [hunk 4]
```diff
@@ -848,5 +850,46 @@ def rec_makedir(target):
...
+    '''
+
+    tar = tarfile.open(tarname, "w:gz")
+    if fnames:
+        for element in fnames:
```
Comment (mturilli) [inline src/radical/utils/misc.py]: Should we check that the file indeed exist? Should we tell the user when it does not? I am thinking about unavoidable typos in the file name.

Comment (andre-merzky) [reply src/radical/utils/misc.py]: the underlying tar module will raise a suitable exception already, so we don't need to replicate that test, fortunately.  I'll add a comment.

**Comment by andre-merzky** [comment]
Thanks for the typos!  I changed the format to `bz2`.

**Pull request merged**